salt/cli/support/collector.py

```diff
@@ -0,0 +1,537 @@
+            and self.config.get("support_archive")
+            and os.path.exists(self.config["support_archive"])
+        ):
+            self.out.warning("Terminated earlier, cleaning up")
+            os.unlink(self.config["support_archive"])
+
+    def _check_existing_archive(self):
+        """
+        Check if archive exists or not. If exists and --force was not specified,
+        bail out. Otherwise remove it and move on.
+
+        :return:
+        """
+        if os.path.exists(self.config["support_archive"]):
+            if self.config["support_archive_force_overwrite"]:
+                self.out.warning(
+                    "Overwriting existing archive: {}".format(
+                        self.config["support_archive"]
+                    )
+                )
+                os.unlink(self.config["support_archive"])
+                ret = True
+            else:
+                self.out.warning(
+                    "File {} already exists.".format(self.config["support_archive"])
+                )
+                ret = False
+        else:
+            ret = True
+
+        return ret
+
+    def run(self):
+        exit_code = salt.defaults.exitcodes.EX_OK
+        self.out = salt.cli.support.console.MessagesOutput()
+        try:
+            self.parse_args()
+        except (Exception, SystemExit) as ex:  # pylint: disable=broad-except
+            if not isinstance(ex, exceptions.SystemExit):
+                exit_code = salt.defaults.exitcodes.EX_GENERIC
+                self.out.error(ex)
+            elif isinstance(ex, exceptions.SystemExit):
+                exit_code = ex.code
+            else:
+                exit_code = salt.defaults.exitcodes.EX_GENERIC
+                self.out.error(ex)
+        else:
+            if self.config["log_level"] not in ("quiet",):
+                self.setup_logfile_logger()
+                salt.utils.verify.verify_log(self.config)
+                salt.cli.support.log = log  # Pass update logger so trace is available
+
+            if self.config["support_profile_list"]:
+                self.out.put("List of available profiles:")
+                for idx, profile in enumerate(
+                    salt.cli.support.get_profiles(self.config)
+                ):
+                    msg_template = "  {}. ".format(idx + 1) + "{}"
+                    self.out.highlight(msg_template, profile)
+                    exit_code = salt.defaults.exitcodes.EX_OK
+            elif self.config["support_show_units"]:
+                self.out.put("List of available units:")
+                for idx, unit in enumerate(self.find_existing_configs(None)):
+                    msg_template = "  {}. ".format(idx + 1) + "{}"
+                    self.out.highlight(msg_template, unit)
+                exit_code = salt.defaults.exitcodes.EX_OK
+            else:
+                if not self.config["support_profile"]:
+                    self.print_help()
+                    raise SystemExit()
```

Comment:
Why not `self.exit()`?

Reply:
I've checked other cli tools and they use `SystemExit` as well as `self.exit()` in almost 50/50 cases. I'm not sure why.

Reply:
Ok.